pkg/deployments/src/task.ts

```diff
@@ -92,6 +92,7 @@ export default class Task {
     const output = this.output({ ensure: false });
     if (force || !output[name]) {
       const instance = await this.deploy(name, args, from, libs);
+      await saveContractDeploymentTransactionHash(instance.address, instance.deployTransaction.hash, this.network);
```

Comment:
This should be in the `deploy` function (which is called here in the previous line), along with the call to `save`. Otherwise, tasks that use `deploy` directly won't save their hashes.

Reply:
Ah true, I was working on the understanding that `deploy` still didn't write the result to `output` so this wasn't necessary.